RecipeApp/AppCoordinator/ApplicationCoordinator.swift

```diff
@@ -0,0 +1,32 @@
+//
+//  ApplicationCoordinator.swift
+//  RecipeApp
+//
+//  Created by Hadley Igoe on 2019-12-09.
+//  Copyright © 2019 Hadley Igoe. All rights reserved.
+//
+
+import UIKit
+
+class ApplicationCoordinator: Coordinator {
+    var navigationController: UINavigationController
+    
+    init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+    
+    func start() {
+        let recipeListViewModel = RecipeListViewModel()
+        let recipeListViewController = RecipeListViewController.create(viewModel: recipeListViewModel)
+        recipeListViewModel.completion = { [weak self] meal in
+            self?.viewRecipeDetails(meal: meal)
+        }
+        navigationController.pushViewController(recipeListViewController, animated: false)
+    }
+    
+    func viewRecipeDetails(meal: Meal) {
+        let recipeDetailsViewController = RecipeDetailsCoordinator(navigationController: navigationController)
```

Comment:
This is not a view controller! And why not just put the meal into the initializer instead of setting a property?